diff --git a/crates/groupware/src/scheduling/organizer.rs b/crates/groupware/src/scheduling/organizer.rs
index 74355b4f..26d28e15 100644
--- a/crates/groupware/src/scheduling/organizer.rs
+++ b/crates/groupware/src/scheduling/organizer.rs
@@ -353,6 +353,7 @@ pub(crate) fn organizer_request_full(
         // Add attendees
         for attendee in &comp.attendees {
-            if (is_first_request && attendee.send_invite_messages())
-                || (!is_first_request && attendee.send_update_messages())
+            if attendee.email.email != itip.organizer.email.email
+                && ((is_first_request && attendee.send_invite_messages())
+                    || (!is_first_request && attendee.send_update_messages()))
             {
                 recipients.insert(&attendee.email.email);
