diff --git a/crates/imap/src/op/store.rs b/crates/imap/src/op/store.rs
index f1c9b09..0b02f44 100644
--- a/crates/imap/src/op/store.rs
+++ b/crates/imap/src/op/store.rs
@@ -7,7 +7,6 @@
 use super::{FromModSeq, ImapContext};
 use crate::{
     core::{SelectedMailbox, Session, SessionData},
-    spawn_op,
 };
 use ahash::AHashSet;
 use common::{network::SessionStream, storage::index::ObjectIndexBuilder};
@@ -52,13 +51,13 @@ impl<T: SessionStream> Session<T> {
         let (data, mailbox) = self.state.select_data();
         let is_condstore = self.is_condstore || mailbox.is_condstore;

-        spawn_op!(data, {
-            let response = data
-                .store(arguments, mailbox, is_uid, is_condstore, op_start)
-                .await?;
+        let response = data
+            .store(arguments, mailbox, is_uid, is_condstore, op_start)
+            .await?;

-            data.write_bytes(response).await
-        })
+        data.write_bytes(response).await;
+
+        Ok(())
     }
 }
